src/Agent.Worker/Container/ContainerInfo.cs

```diff
@@ -235,6 +235,9 @@ public void ExpandProperties(Variables variables)
                 // After mount volume variables are expanded, they are final
                 MountVolumes.Add(new MountVolume(volume));
             }
+
+            // Expand env vars
+            variables.ExpandValues(ContainerEnvironmentVariables);
```

Comment:
cool, thanks for adding this.
Do we need to do this for `ContainerCreateOptions`?
also, do we need to do this for the job container?

Reply:
It applies to job containers too. See: https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/JobRunner.cs#L228 

I added expansion on the options string, and also just added it for image too